src/common/meson.build

```diff
@@ -0,0 +1,400 @@
+    config_h.set10('TUKLIB_CPUCORES_PSTAT_GETDYNAMIC', true)
+endif
+
+# tuklib_integer.m4
+
+bswap_src = '''
+int main()
+{
+	__builtin_bswap16(1);
+	__builtin_bswap32(1);
+	__builtin_bswap64(1);
+	return 0;
+}
+'''
+res = cc.links(bswap_src, name: '__builtin_bswap16/32/64')
+if res
+    config_h.set10('HAVE___BUILTIN_BSWAPXX', true,
+        description: 'Define to 1 if the GNU C extensions __builtin_bswap16/32/64 are supported.'
+        )
+else
+    have_byteswap_h = false
+    have_sys_endian_h = false
+    have_sys_byteorder_h = false
+    bswap_headers = [
+        'byteswap.h',
+        'sys/endian.h',
+        'sys/byteorder.h'
+    ]
+    foreach h : bswap_headers
+        if cc.has_header(h)
+            config_h.set10('HAVE_' + h.underscorify().to_upper(), true)
+            set_variable(f'have_' + h.underscorify(), true)
+        endif
+    endforeach
+
+    if have_byteswap_h == true
+        bswap_fcts = [
+            'bswap_16',
+            'bswap_32',
+            'bswap_64'
+        ]
+        foreach f : bswap_fcts
+            bswap_src = '''
+#include <byteswap.h>
+int main()
+{
+	@0@(42);
+	return 0;
+}
+'''
+            res = cc.links(bswap_src.format(f), name: f)
+            if res == true
+                config_h.set10('HAVE_' + f.to_upper(), true,
+                    description: 'Define to 1 if ' + f + '() is available.'
+                )
+            endif
+        endforeach
+    endif
+endif
+
+fast_unaligned_access = false
+if get_option('unaligned-access').auto() == true
+    unaligned_access_cpus = [
+        'ppc',
+        'ppc64',
+        'x86',
+        'x86_64',
+    ]
+    foreach cpu : unaligned_access_cpus
+        if host_machine.cpu_family() == cpu
+            config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+                description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+            )
+            fast_unaligned_access = true
+            break
+        endif
+    endforeach
+
+    if fast_unaligned_access == false
+        unaligned_access_cpus = [
+            'aarch64',
+            'arm',
+            'riscv32',
+            'riscv64',
+        ]
+        foreach cpu : unaligned_access_cpus
+            if host_machine.cpu_family() == cpu
+                ua_src = '''
+#if !defined(__ARM_FEATURE_UNALIGNED) \
+		&& !defined(__riscv_misaligned_fast) \
+		&& !defined(_MSC_VER)
+compile error
+#endif
+int main(void) { return 0; }
+'''
+                res = cc.compiles(ua_src, name: 'fast unaligned access for ' + cpu)
+                if res == true
+                    config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+                    description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+                    )
+                    fast_unaligned_access = true
+                    break
+                endif
+            endif
+        endforeach
```

Comment:
```suggestion
        if host_machine.cpu_family() in unaligned_access_cpus
            ua_src = '''
#if !defined(__ARM_FEATURE_UNALIGNED) \
		&& !defined(__riscv_misaligned_fast) \
		&& !defined(_MSC_VER)
compile error
#endif
int main(void) { return 0; }
'''
            if cc.compiles(ua_src, name: 'fast unaligned access for ' + cpu)
                config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
                description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
                )
                fast_unaligned_access = true
            endif
        endif
```

Reply:
maybe the first list too can be improved (note that the variable cpu was used in the description of check line 230 of my original code) :

```diff --git a/src/common/meson.build b/src/common/meson.build
index d01853b1..ed3d3a75 100644
--- a/src/common/meson.build
+++ b/src/common/meson.build
@@ -182,15 +182,12 @@ if get_option('unaligned-access').auto()
         'x86',
         'x86_64',
     ]
-    foreach cpu : unaligned_access_cpus
-        if host_machine.cpu_family() == cpu
-            config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
-                description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
-            )
-            fast_unaligned_access = true
-            break
-        endif
-    endforeach
+    if host_machine.cpu_family() in unaligned_access_cpus
+        config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+            description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+        )
+        fast_unaligned_access = true
+    endif

     if not fast_unaligned_access
         unaligned_access_cpus = [
@@ -199,9 +196,8 @@ if get_option('unaligned-access').auto()
             'riscv32',
             'riscv64',
         ]
-        foreach cpu : unaligned_access_cpus
-            if host_machine.cpu_family() == cpu
-                ua_src = '''
+        if host_machine.cpu_family() in unaligned_access_cpus
+            ua_src = '''
 #if !defined(__ARM_FEATURE_UNALIGNED) \
                && !defined(__riscv_misaligned_fast) \
                && !defined(_MSC_VER)
@@ -209,15 +205,13 @@ compile error
 #endif
 int main(void) { return 0; }
 '''
-                if cc.compiles(ua_src, name: 'fast unaligned access for ' + cpu)
-                    config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
-                    description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
-                    )
-                    fast_unaligned_access = true
-                    break
-                endif
+            if cc.compiles(ua_src, name: 'fast unaligned access for ' + host_machine.cpu_family())
+                config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+                description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+                )
+                fast_unaligned_access = true
             endif
-        endforeach
+        endif
     endif
 elif get_option('unaligned-access').enabled()
     config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
```